src/controller/python/controller.i

```diff
@@ -84,6 +84,8 @@ using namespace std;
 //----------------------------------------------------------------------------------------------
 
 %thread webots::Robot::step(int duration);
+%thread webots::Robot::step_begin(int duration);
```

Comment:
Is this mandatory ? I found no documentation about the SWIG `%thread` function

Reply:
The origin of that is in #3389. Some explanation is provided [here](http://www.swig.org/Doc4.0/Python.html).

Reply:
It seems that all functions must be locked to avoid problems with multi-threaded python controllers. Only the step function can run in parallel. So it makes sense to enable multithreading on the 2 new API functions.